src/libsyntax/parse/parser.rs

```diff
@@ -439,6 +442,7 @@ impl<'a> Parser<'a> {
     pub fn new(sess: &'a ParseSess,
                tokens: TokenStream,
                directory: Option<Directory>,
+               recurse_into_file_modules: bool,
```

Comment:
I put it here to be next to the `directory` option since they are related. My first attempt tried to combine the two into an enum, but it was really grotty. Builder would be best I guess, but yeah, later.